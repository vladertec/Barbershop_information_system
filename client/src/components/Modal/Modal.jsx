const Modal = ({ updateModalIsOpen, type }) => {
  return (
    <div className="modal">
      <div className="modal__container">
        <button
          className="modal__button-x"
          onClick={() => {
            updateModalIsOpen(false)
          }}
        >
          X
        </button>
      </div>
      <div className="modal__text-container">
        <h3 className="modal__text">
          Your products have been added to the {type}
        </h3>
        <div>
          <button
            className="modal__button"
            onClick={() => {
              updateModalIsOpen(false)
            }}
          >
            Ok
          </button>
        </div>
      </div>
    </div>
  )
}

export default Modal
