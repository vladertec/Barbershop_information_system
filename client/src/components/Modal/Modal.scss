.modal {
  background-color: #2a2416;
  width: 80%;
  color: white;
  font-weight: bold;
  font-family: "Arima Madurai", serif;

  &__container {
    background-color: #999999;
    text-align: right;
  }

  &__button-x {
    font-size: 27px;
    border: none;
    background-color: rgb(170, 48, 7);
    color: white;
    cursor: pointer;
  }

  &__text-container {
    margin-top: 7%;
    display: flex;
    @include flex-container;
    flex-direction: column;
  }
  &__text {
    display: block;
    text-align: center;
    font-weight: bold;
    font-size: 2vw;
  }

  &__button {
    margin-top: 25%;
    margin-bottom: 25%;
    color: white;
    background-color: #b48e40;
    border-radius: 10px;
    cursor: pointer;
    font-size: 2.5vw;

    &:hover {
      background-color: #7b5d20;
      font-weight: bold;
    }
  }
}
