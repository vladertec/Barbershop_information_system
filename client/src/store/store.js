import { createStore, combineReducers, compose, applyMiddleware } from "redux"
import newsReducer from "./news/reducer"
import barberReducer from "./barbers/reducer"
import productsReducer from "./products/reducer"
import visitReducer from "./visits/reducer"
import cartListReducer from "./cart/reducer"
import favouritesListReducer from "./favourite/reducer"
import thunk from "redux-thunk"

const syncCartListLS = (store) => {
  return function (next) {
    return function (action) {
      if (
        action.type === "ADD_PRODUCT_IN_CART" ||
        action.type === "REMOVE_PRODUCT_FROM_CART" ||
        action.type === "REMOVE_ALL_PRODUCT_FROM_CART"
      ) {
        const result = next(action) // starts reducer
        localStorage.setItem(
          "myCart",
          JSON.stringify(store.getState().cart.cartList)
        )
        return result
      }
      return next(action)
    }
  }
}

const syncFavouriteListLS = (store) => {
  return function (next) {
    return function (action) {
      if (
        action.type === "ADD_FAVOURITE" ||
        action.type === "REMOVE_FAVOURITE"
      ) {
        const result = next(action) // starts reducer
        localStorage.setItem(
          "myFavorites",
          JSON.stringify(store.getState().favouriteList)
        )
        return result
      }
      return next(action)
    }
  }
}

export const reducer = combineReducers({
  news: newsReducer,
  products: productsReducer,
  barber: barberReducer,
  visit: visitReducer,
  cart: cartListReducer,
  favourite: favouritesListReducer,
})

const devTools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f

const store = createStore(
  reducer,
  compose(applyMiddleware(thunk, syncCartListLS, syncFavouriteListLS), devTools)
)

export default store
