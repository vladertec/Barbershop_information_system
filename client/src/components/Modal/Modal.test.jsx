import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { Provider } from "react-redux"
import { legacy_createStore as createStore } from "redux"
import { reducer } from "../../store/store"
import Modal from "./Modal"
import {shallow } from 'enzyme';

describe("<Modal /> component", () => {
  it("should render", () => {
    const store = createStore(reducer)
    render(
      <Provider store={store}>
        <Modal />
      </Provider>
    )
  })

  it("it should show button text", () => {
    const store = createStore(reducer)
    render(
      <Provider store={store}>
        <Modal />
      </Provider>
    )
    screen.getByText(/Ok/i)
  })

  it("do snapshot", () => {
    const store = createStore(reducer)
    const { container } = render(
      <Provider store={store}>
        <Modal />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })

  // it("should pass onClick if button is clicked", () => {
  //   const store = createStore(reducer)
  //   const { container, debug } = render(
  //     <Provider store={store}>
  //       <Modal />
  //     </Provider>
  //   )
  //   userEvent.pointer({ keys: "[MouseLeft]", target: screen.getByText("Ok") })
  //   expect(jest.fn()).toHaveBeenCalledTimes(1)
  // })
})
