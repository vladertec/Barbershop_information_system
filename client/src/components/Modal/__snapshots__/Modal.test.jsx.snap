// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component do snapshot 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="modal__container"
    >
      <button
        class="modal__button-x"
      >
        X
      </button>
    </div>
    <div
      class="modal__text-container"
    >
      <h3
        class="modal__text"
      >
        Your products have been added to the 
      </h3>
      <div>
        <button
          class="modal__button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>
</div>
`;
